src/core/Akka.Remote/RemoteSystemDaemon.cs

```diff
@@ -211,9 +223,16 @@ private void TerminationHookDoneWhenNoChildren()
         protected override void TellInternal(object message, IActorRef sender)
         {
             OnReceive(message);
+            Sender = sender ?? ActorRefs.Nobody;
         }
 
         /// <summary>
+        /// For system messages that require a reply back to the <see cref="IActorRef"/> responsible
+        /// for telling a message to the <see cref="RemoteSystemDaemon"/>.
+        /// </summary>
+        protected IActorRef Sender { get; private set; }
```

Comment:
I checked the scala version and they do not do this, they handle all sender related stuff inside their huge `!` tell handler.